packages/site/src/lib/components/Slides.svelte

```diff
@@ -1,35 +1,66 @@
 <script lang="ts">
   export let events: Event[];
+
   import Image from "$lib/components/Image.svelte";
+  import ChevronRight from "@material-symbols/svg-400/rounded/chevron_right.svg?component";
+  import ChevronLeft from "@material-symbols/svg-400/rounded/chevron_left.svg?component";
+
+  interface Poster {
+    altText: string;
+    imageUrl: string;
+  }
 
   interface Event {
-    poster?: {
-      altText: string;
-      imageUrl: string;
-    };
+    poster?: Poster
+  }
+
+  const posters = (): Poster[] => events.flatMap((ev) => ev.poster ? [ev.poster] : [])
```

Comment:
```suggestion
  $: posters: Poster[] = events.flatMap((ev) => ev.poster ? [ev.poster] : [])
```

This should use the [reactive declaration syntax](https://svelte.dev/tutorial/reactive-declarations) so Svelte knows to update it when the props update (props are reactive and Svelte might share a component instance and just swap props).

Also TIL flatMap can be used as a filtermap!!

Reply:
ha yea never `.filter().map()`